source/proxy_connection.c

```diff
@@ -1085,6 +1129,95 @@ int aws_http_client_connect_via_proxy(const struct aws_http_client_connection_op
     }
 }
 
+static int s_connect_proxy_via_env_variable(const struct aws_http_client_connection_options *options) {
+    struct aws_http_proxy_options proxy_options;
+    AWS_ZERO_STRUCT(proxy_options);
+    struct aws_uri proxy_uri;
+    AWS_ZERO_STRUCT(proxy_uri);
+    struct aws_tls_connection_options default_tls_connection_options;
+    AWS_ZERO_STRUCT(default_tls_connection_options);
+    bool found = false;
+    bool success = false;
+    if (s_proxy_uri_init_from_env_variable(options->allocator, options, &proxy_uri, &found)) {
+        /* Envrionment is set but failed to parse it */
+        goto done;
+    }
+    if (found) {
+        proxy_options.host = proxy_uri.host_name;
+        proxy_options.port = proxy_uri.port;
+        proxy_options.connection_type = options->proxy_ev_settings->connection_type;
+        if (!proxy_options.connection_type) {
+            if (options->tls_options) {
+                /* Use Tunneling when main connection use TLS. */
+                proxy_options.connection_type = AWS_HPCT_HTTP_TUNNEL;
+            } else {
+                /* Use forwarding proxy when main connection use clear text. */
+                proxy_options.connection_type = AWS_HPCT_HTTP_FORWARD;
+            }
+        }
+        if (aws_byte_cursor_eq_ignore_case(&proxy_uri.scheme, &aws_http_scheme_https)) {
+            if (options->proxy_ev_settings->tls_options) {
+                proxy_options.tls_options = options->proxy_ev_settings->tls_options;
+            } else {
+                struct aws_tls_ctx *tls_ctx = NULL;
+                struct aws_tls_ctx_options tls_ctx_options;
+                AWS_ZERO_STRUCT(tls_ctx_options);
+                /* create a default tls options */
+                aws_tls_ctx_options_init_default_client(&tls_ctx_options, options->allocator);
+                /* turn off verify peer for proxy connection setup by default */
+                aws_tls_ctx_options_set_verify_peer(&tls_ctx_options, false);
```

Comment:
This is very questionable.  I imagine it's necessary for any integration tests to pass (since our tls proxy integration tests use a proxy that presents a signed localhost certificate which isn't going to pass server name validation) but for people using this in the wild with a real properly-set-up proxy, server name validation is critical.  I'm not sure how to resolve this, but we can't default it to off.

Reply:
One possibility is to allow for an environment variable to disable server name validation and set that in the integration test environment setup script.  It means we don't have a test for the "real" path where server name validation is done though, but that may be a TBD.

Reply:
Yeah! Totally make sense. Added another env variable. Set it up for tests as well.